### ISSUE TITLE
Problem compiling app for web from version 7.6.0

### ISSUE THREAD

**Issue by mklepaczko** [781flyingdutchman/background_downloader#124]
I built an app that is published for Android, IOS and web. The downloading functions are enabled on IOS and Android.

Till I was using v7.4.1 I could compile the app to all platforms, when upgrading to 7.50 it still works but starting from version 7.6.0 
 I get an error when compiling to web:

: Error: Member not found: 'BackgroundIsolateBinaryMessenger.ensureInitialized'.
desktop_downloader_isolate.dart:37
  BackgroundIsolateBinaryMessenger.ensureInitialized(rootIsolateToken);

                                   ^^^^^^^^^^^^^^^^^
Failed to compile application.

It seems that even if the package is not used it tries to initialise the isolate, which is no possible on web.

Hence I can not build the app anymore for web event thought I do not use the the downloader functionnality, and I expect to keep one code base for all platforms.

[✓] Flutter (Channel stable, 3.10.5, on macOS 13.4.1 22F770820d darwin-arm64, locale pl-PL)
    • Flutter version 3.10.5 on channel stable at /Users/michalklepaczko/projekty_dev/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 796c8ef792 (3 months ago), 2023-06-13 15:51:02 -0700
    • Engine revision 45f6e00911
    • Dart version 3.0.5
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/michalklepaczko/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] VS Code (version 1.81.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 0E105A89-5361-420A-AD0D-53361F30B6D7 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 13.4.1 22F770820d darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 116.0.5845.110

[✓] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by 781flyingdutchman** [comment]
Duplicate of #123. Looks liek this requires a conditional import - not sure how to do this, but I'll do some research.

**Comment by 781flyingdutchman** [comment]
The issue is slightly different though: `BackgroundIsolateBinaryMessenger.ensureInitialized` is not the same as `WidgetsFlutterBinding.ensureInitialized` and - importantly - does not compile at all when using web. The conditional import I was referring to is therefore not about _you_ doing a conditional import, but _me_ doing a conditional import such that - when compiling for web - the `BackgroundIsolateBinaryMessenger.ensureInitialized` code is not imported.  I think that can be done, but I have to find some time to look into this and try it out.

**Comment by mklepaczko** [comment]
Ok, sorry, I did not undestand that conditionnal import was on your side.
Hope there is a way, for now I will stick to 7.4.1

**Comment by 781flyingdutchman** [comment]
Hi, can you test a development branch to see if you can compile for the web?  Change your pubspec.yaml to:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: web
```

and let me know if it works. It's just a stub implementation, no actual web functionality, so we're just testing for compile errors here.

**Comment by mklepaczko** [comment]
Yes this version does build on web :)

**Comment by jamescardona11** [comment]
It is working for me as well

**Comment by 781flyingdutchman** [comment]
Published in V7.9.4

**Comment by mklepaczko** [comment]
Hi, thanks for this fast update:) 
I would have an issue related to another project, could you contact me on my email to discuss it? That might me a quick thing for you to do and my client is willing to pay for that, it concenrs another pluggin and IOS dev is needed. michal.klepaczko@gmail.com. Thx best.
